### ISSUE TITLE
Different errors with same version on Github and locally?

### ISSUE THREAD

**Issue by ivnsch** [rust-lang/rust-clippy#8515]
### Description

Running Clippy as part of Github CI (Ubuntu 20.04.3) is showing a few errors that I don't see with my local installation (macOS 12.1). 
Both are using the same versions:

```
clippy -V
clippy 0.1.59 (9d1b210 2022-02-23)

rustc -V
rustc 1.59.0 (9d1b2106e 2022-02-23)

cargo -V
cargo 1.59.0 (49d8809dc 2022-02-10)

rustup -V
rustup 1.24.3 (ce5817a94 2021-05-31)
```

The errors are a pretty old check:
```rust
res.to_u64().ok_or(anyhow!("...")) // try this: `ok_or_else(|| ...`
```

The command is also the same:
```
cargo clippy -- -D warnings
```
And it's being ran from the repo's root (both locally and on Github).

I cleaned etc. locally but still nothing. This is probably less about the Github CI than my local installation just not showing these errors, any ideas?

Note also that my local installation showed some errors of the same type (`ok_or` -> `ok_or_else`) in other parts of the repo.

### Version

```text
rustc 1.59.0 (9d1b2106e 2022-02-23)
binary: rustc
commit-hash: 9d1b2106e23b1abd32fce1f17267604a5102f57a
commit-date: 2022-02-23
host: aarch64-apple-darwin
release: 1.59.0
LLVM version: 13.0.0
```

**Comment by xFrednet** [comment]
Hey, could you maybe provide the command that you use locally and in your CI? It might also be helpful to have the actual output log. Clippy has very few conditional lints based on the target architecture, but I'm guessing that you don't refer to those lints. :upside_down_face:

**Comment by matiasg** [comment]
I am having a similar problem in gitlab.

### output in gitlab
```
$ rustc --version
rustc 1.66.0 (69f9c33d7 2022-12-12)
$ cargo --version
cargo 1.66.0 (d65d197ad 2022-11-15)
$ rustup component add clippy
info: downloading component 'clippy'
info: installing component 'clippy'
$ cargo clippy -- -D warnings
    Updating crates.io index
 Downloading crates ...
  Downloaded num_threads v0.1.6
  Downloaded time-core v0.1.0
  Downloaded time-macros v0.2.6
  Downloaded rand v0.8.5
 .....
error: use of deprecated struct `chrono::Date`: Use `NaiveDate` or `DateTime<Tz>` instead
 --> src/j2_predictors.rs:3:14
  |
3 | use chrono::{Date, DateTime, Datelike, Duration, Timelike, Utc};
  |              ^^^^
  |
  = note: `-D deprecated` implied by `-D warnings`
.... similar errors ....
error: could not compile `rostit-predictor` due to 7 previous errors
```

### locally
```
root@9593fa576b02:/# cd /code
root@9593fa576b02:/code# rustc --version
rustc 1.66.0 (69f9c33d7 2022-12-12)
root@9593fa576b02:/code# cargo --version
cargo 1.66.0 (d65d197ad 2022-11-15)
root@9593fa576b02:/code# rustup component add clippy
info: downloading component 'clippy'
info: installing component 'clippy'
root@9593fa576b02:/code# cargo clippy -- -D warnings
    Updating crates.io index
  Downloaded time v0.1.44
  Downloaded slog-term v2.9.0
....
    Checking rostit-predictor v0.1.0 (/code)
    Finished dev [unoptimized + debuginfo] target(s) in 1m 40s
```
and no warnings/errors are shown.

**Comment by matiasg** [comment]
I was having the same problem but I found out that I had different versions of a crate:
`chrono v0.4.23` vs `chrono v0.4.22`

**Comment by CommanderStorm** [comment]
I don't think this is an issue with clippy.
I suspect this might be related to `--locked` (which is not included in the command above), not using lockfiles or something, but not sure. defintively not reproducible or actionable at this point in time.

We can reopen if this changes, or discuss in an new issue with an repro ^^
